Merged content from two articles into one (#5985)
umbraco/UmbracoDocs: 14/umbraco-cms/tutorials/editors-manual/working-with-content/README.md

| Ctrl + Y | Redo       |
| Ctrl + Z | Undo       |

We have listed only a few keyboard shortcuts. For a detailed list of available keyboard shortcuts, see the [TinyMCE Documentation](https://www.tiny.cloud/docs/tinymce/latest/keyboard-shortcuts/).

> ## View Source Code

Author: I have fixed the heading tags. H2 for all items displayed on RTE and all other in depth content (add/edit/delete) as H3 tags,

![View Source Code](images/view-source-code-v11.png)

If you are proficient in HTML, you can switch to HTML mode to create your page. You can also check the code and make minor adjustments to get the page exactly as you want.

Certain elements such as scripts are not recognized by the HTML view of the Rich Text Editor. You can enter the scripts directly in the text view of the editor.

### Text Formatting

You do not normally need to spend much time formatting text because Umbraco takes care of the formatting. However, the editor provides some options for controlling the text styles.

> ### Formats

Author: Moved it above text formatting section.

{% endhint %}

#### Remove Formatting

![Remove Format Button](images/Remove-Format-v11.png)

If you have already formatted a paragraph or selection using the formatting buttons, you can remove the formatting rule.

To remove formatting:

1. Select the text you want to remove the style from.
2. Click the relevant formatting button to remove the formatting rule.

You can also add a **Remove format** button in your toolbar. To add the **Remove format** button:

1. Navigate to your Rich Text Editor in the Document Type.
2. Click the cog wheel.
3. Click **Edit** next to the Rich Text Editor Data Type.
4. Select **Remove format** under the **Toolbar Configuration**.
5. Click **Submit**.
6. Click **Save**.

### Links

![Link Button](images/Link-Button-v11.png)

The **Insert/Edit Link** button is used to add or update links to internal pages, external pages, media files, email links, and anchors. The process for inserting a hyperlink differs depending on the type of hyperlink you wish to create:

To insert different types of hyperlinks, follow these steps:

> #### Link to a Page on Another Website

Author: yup, makes sense 👍will add the content in dropdowns.

![Linking to an anchor](images/Linking-to-Anchor-v11.png)

### Create a Link from an Image

You can make images into clickable links in Umbraco:

1. Insert an image into the Rich Text Editor.
   - For more information, see the [Working with Images](#working-with-images) section.
2. Select the image that will form the hyperlink.
3. Enter the URL of the web page you wish to link to in the **Link** field.
4. Enter the text that will be displayed as the link title in the **Link Title** field.
5. Select the **Target** field to open the link in a new window or tab.
6. Click **Submit**.

![Create a Link from an Image](images/Link-from-Image-v11.png)

#### Removing a Link

![Remove link Button](images/Remove-link-v11.png)

To remove a link:

1. Select the link in the Rich Text Editor.
   - For text links, click the cursor anywhere within the link text. For an image, click the image itself.
2. Click the **Remove Link** button which will remove the hyperlink.
3. Alternatively, you can click the **Insert/Edit Link** button and remove the link from the **Link** field.

### Working with Images

To display images on a page the images must be uploaded to your Umbraco media library.

> Many organizations set up a media library containing images that editors can use on their pages.  Others allow their editor's free use of their images. The procedure for uploading an image varies slightly depending on which method your organization uses. Check with your system administrator about what applies to your organization.

Reviewer: ```suggestion
Many administrators set up a media library containing images that editors can use on their pages.  Others allow their editor's free use of their images. The procedure for uploading an image varies slightly depending on which method your administrators have setup. Check with your system administrator for more information about this.
```

#### Inserting an Image from the Media Library

1. Place the cursor in the Rich Text Editor where you want to insert your image.
2. Click the **Media Picker** button from the toolbar.
3. Select the folder in which the image is.
4. Click the thumbnail of your chosen image to open the image properties menu.
5. Enter a name/description for the image in the **Caption (optional)** field.
   - It is important to add descriptive titles to images as these are used to assist visually impaired users.
6. Click **Select**.

![Inserting an Image from the Media Library](images/Inserting-Image-from-the-Media-Library-v11.png)

#### Inserting an Image from your Computer

> You can upload images directly from the Rich Text Editor on the page you are editing. These images will be stored in the Umbraco Media Library. Therefore care should be taken to ensure that the image is placed in the correct location within the library. If you click the plus icon underneath the search bar in the media picker slide-out menu you can create folders in the media library.

Reviewer: ```suggestion
You can upload images directly from the Rich Text Editor on the page you are editing. These images will be stored in the Umbraco Media Library. Therefore, it would be best to ensure the image is placed in the correct location within the library. If you click the plus icon underneath the search bar in the media picker slide-out menu you can create folders in the media library.
```

#### Editing Macros

You can edit values after a macro has been added to the Rich Text Editor:

1. Select the macro in the Rich Text Editor.
2. Click the **Insert Macro** button from the toolbar.
   - This will open the configuration menu where you can edit any values previously added.

### Removing macros

To remove a macro:

1. Select the macro in the Rich Text Editor.
2. Press the **Delete** or **Backspace** button on your keyboard.

### Tables

Tables are used to format information in a grid-based structure. When you insert a table, you select how many rows and columns the table should comprise of. Additionally, you can fill in some optional formatting properties. These values can be changed later, so it is not important to know exactly what your table will look like when you create it.

![Inserting a Table](images/Insert-a-table-v11.png)

#### Editing an Existing Table

To edit the table after creating it, click on the table. A pop-up appears with different table properties and options. Alternatively, you can click on the **Table** button in the Rich Text Editor toolbar.

![Editing an Existing Table](images/Editing-an-existing-table-v11.png)

> Clicking on **Table Properties** gives you different options for modifying the table’s appearance. However, the developer of the website may have already created table styles for you so you may not need to adjust these settings. There are other options available for modifying cells, rows, and columns such as width, height, alignment, border, and so on.

Reviewer: ```suggestion
Clicking on **Table Properties** gives you different options for modifying the table’s appearance. However, the developer of the website may have already created table styles for you so you may not need to adjust these settings. 

There are other options available for modifying cells, rows, and columns such as width, height, alignment, border, and so on.
```
